### ISSUE TITLE
Switch to Cached API Client

### ISSUE THREAD

**Issue by lasuk** [macxred/cashctrl_ledger#12]
**Problem:**
Tests to mirror VAT codes and account numbers take too long due to repeated API calls.

**Solution:**
I've proposed to extend the API client and cache accounts and tax codes locally to avoid repeated lookups in macxred/cashctrl_api#26. Once this feature is implemented, we can use the cached version of the API client.

**Implementation Steps:**
Use `CachedCashCtrlClient` instead of `CashCtrlClient` in `CashCtrlLedger`. This involves:
- Switch type of `self._client` from `CashCtrlClient` to `CachedCashCtrlClient`.
- Extend mutator methods to invalidate cache with below methods when making changes on the remote.
    ```
    invalidate_accounts_cache()
    invalidate_tax_codes_cache()
    ```
- Use the new lookup functions when mapping to/from account id or tax id:
    ```
    account_to_id(account: float) -> int
    account_from_id(id: int) -> float
    tax_code_to_id(code: str) -> int
    tax_code_from_id(id: int) -> str
    ```

**Comment by lasuk** [comment]
Let's extend the cache to currencies, and use the additional lookup functions from/to currency id:
```
currency_to_id(code: str) -> int
currency_from_id(id: int) -> str
```

**Comment by AlexTheWizardL** [comment]
Further entities that should have cache:
- account_category
- file
- ledger
- ledger_category - maybe

**Comment by lasuk** [comment]
Implemented in #32.
